### PR TITLE
ARROW-12395: Create RunInSerialExecutor benchmark

### PR DESCRIPTION
**Pull request by westonpace**

### CODE REVIEW

File: cpp/src/arrow/util/thread_pool_benchmark.cc [hunk 1]
```diff
@@ -103,6 +103,22 @@ static void ThreadPoolSpawn(benchmark::State& state) {  // NOLINT non-const refe
...
+// Benchmark SerialExecutor::RunInSerialExecutor
+static void RunInSerialExecutor(benchmark::State& state) {  // NOLINT non-const reference
+  const auto workload_size = static_cast<int32_t>(state.range(0));
+
+  Workload workload(workload_size);
+
+  for (auto _ : state) {
+    SerialExecutor::RunInSerialExecutor<arrow::detail::Empty>(
+        [&](internal::Executor* executor) {
+          return DeferNotOk(executor->Submit(workload));
```
Comment (pitrou) [inline cpp/src/arrow/util/thread_pool_benchmark.cc:115]: Use `std::ref(workload)` for consistency with other benchmarks?

**Comment by westonpace** [comment]
Failing CI looks to be a dockerhub issue.  Can someone restart the actions here?

**Comment by pitrou** [comment]
The easiest to restart them all (rather than clicking on each of them) is to force-push this branch, I think.

**Comment by westonpace** [comment]
JNI failure is ARROW-11633 and unrelated.

**Review by pitrou — APPROVED** [review]
+1, thank you!

**Pull request closed without merging**